Data/Vector.hs

```diff
@@ -381,6 +382,24 @@ instance MonadZip Vector where
   {-# INLINE munzip #-}
   munzip = unzip
 
+instance MonadFix Vector where
+  -- We take care to dispose of v0 as soon as possible.
+  -- We also avoid setting up the result vector to refer to
+  -- itself. These measures should prevent memory leaks.
+  -- It's perfectly safe to use non-monadic indexing within
+  -- each element, as the result of indexing will be demanded
+  -- as soon as the vector is produced.
```

Comment:
What I mean about this is that the use of `(!)` in line 399 is okay: it can't cause a vector to be kept alive by a thunk. Why? Because we don't *create* that vector until we need the value of the result of that indexing operation. This is as opposed to the `headM` we need to get `h`, where we'd otherwise keep `v0` alive in the first element of the result vector until that first element is forced by someone else.

Reply:
Thanks! 

I'll integrate that explanation into comment tomorrow. Right now it's somewhat cryptic